www/main/share-sync.js

```diff
@@ -19,26 +24,44 @@ module.exports = {
   },
   async tryConnection() {
     const host = store.get('userId');
+    let syncCode = null;
 
-    if (window.namespaceString) {
-      return window.namespaceString;
-    }
-    try {
-      const result = await request(`${SYNC_API_URL}/sync/begin/${host}`);
-      const {
-        data: { namespaceString },
-      } = JSON.parse(result);
-
-      if (window.io !== undefined) {
-        window.namespaceString = namespaceString;
-        onConnect(namespaceString);
-      } else {
-        // TODO: Wait for io or something
+    const getNewCode = async () => {
+      let newCode = null;
+      try {
+        const result = await request(`${SYNC_API_URL}/sync/begin/${host}`);
+        const {
+          data: { namespaceString },
+        } = JSON.parse(result);
+
+        if (window.io !== undefined) {
+          window.namespaceString = namespaceString;
+          onConnect(namespaceString);
+        }
+
+        newCode = namespaceString;
+      } catch (error) {
+        analytics.trackEvent('sync', 'error', error);
+        new Noty({
+          type: 'error',
+          text: i18n.t('TOOLBAR.SYNC_CONTROLLER.CODE_ERR'),
+          timeout: 3000,
+          modal: true,
+        }).show();
+        newCode = null;
       }
-      return namespaceString;
+      return newCode;
+    };
+
+    // if a succesful code already exists, use that or else get new code
+    try {
+      await request(`${SYNC_API_URL}/sync/join/${window.namespaceString}`);
+      syncCode = window.namespaceString;
     } catch (e) {
-      return false;
+      syncCode = getNewCode();
```

Comment:
do you need to try the join call again?

Reply:
Yes, so when the existing code has stopped working we need to get a new code.